mne/tests/test_cov.py

```diff
@@ -544,13 +531,10 @@ def get_data(n_samples, n_features, rank, sigma):
                  sigma=sigma)
     method_params = {'iter_n_components': [n_features + 5]}
     msg = ('You are trying to estimate %i components on matrix '
-           'with %i features.')
-    with warnings.catch_warnings(record=True) as w:
-        warnings.simplefilter('always')
+           'with %i features.') % (n_features + 5, n_features)
+    with pytest.warns(RuntimeWarning, match=msg):
```

Comment:
maybe msg is too much and its more readable to use match

Reply:
I'm trying to make minimal changes to get us to the new format with this PR, I'd rather leave stuff like this for a follow up if someone is motivated. Things can always be made better, so I'd rather stick with ensuring the changes in this PR do that

Reply:
(as opposed to making sure they make them optimal)